rust/agama-logs/src/main.rs

```diff
@@ -0,0 +1,215 @@
+extern crate fs_extra;
+extern crate tempdir;
+
+use fs_extra::copy_items;
+use fs_extra::dir::CopyOptions;
+use nix::unistd::Uid;
+use std::fs;
+use std::fs::File;
+use std::io;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tempdir::TempDir;
+
+const DEFAULT_COMMANDS: [&str; 3] = [
+    "journalctl -u agama",
+    "journalctl -u agama-auto",
+    "journalctl --dmesg",
+];
+
+const DEFAULT_PATHS: [&str; 14] = [
+    // logs
+    "/var/log/YaST2",
+    "/var/log/zypper.log",
+    "/var/log/zypper/history*",
+    "/var/log/zypper/pk_backend_zypp",
+    "/var/log/pbl.log",
+    "/var/log/linuxrc.log",
+    "/var/log/wickedd.log",
+    "/var/log/NetworkManager",
+    "/var/log/messages",
+    "/var/log/boot.msg",
+    "/var/log/udev.log",
+    // config
+    "/etc/install.inf",
+    "/etc/os-release",
+    "/linuxrc.config",
+];
+
+const DEFAULT_RESULT: &str = "/tmp/agama_logs";
+const DEFAULT_NOISY: bool = true;
+const DEFAULT_COMPRESSION: (&str, &str) = ("bzip2", "tar.bz2");
+const DEFAULT_TMP_DIR: &str = "agama-logs";
+
+// A wrapper around println which shows (or not) output depending on noisy boolean variable
+macro_rules! showln
+{
+    ($n:expr, $($arg:tt)*) => { if($n) { println!($($arg)*) } }
+}
+
+// A wrapper around println which shows (or not) output depending on noisy boolean variable
+macro_rules! show
+{
+    ($n:expr, $($arg:tt)*) => { if($n) { print!($($arg)*) } }
+}
+
+// Struct for log represented by a file
+struct LogPath {
+    // log source
+    src_path: &'static str,
```

Comment:
There was a problem with strings with "not living long enoug" (or something like that), so I switched to ```str```. I can give it another try since the code evolved a bit since that time.

Reply:
As it seems this one was turned into ```String``` successfully